### ISSUE TITLE
Cannot deploy forms or launch Enketo with local HTTPS

### ISSUE THREAD

**Issue by Humblehound** [kobotoolbox/kobocat#474]
Hi,
I'm able to setup everything using local config. Once I use the server config, I've identified at least two issues:

- Any attempt to open Enketo from Kobotoolbox fails (preview of a form or opening a submission)
- More importantly, I'm unable to deploy a form (duplicate of [#162](https://github.com/kobotoolbox/kobo-docker/issues/162))

As in #162, I've verified that `koboform_surveydraft` is nonexistent in my database with the use of 

 `SELECT (1) AS "a" FROM "koboform_surveydraft" LIMIT 1`. The postgres log contains the following errors:
> 2018-08-08 14:32:59 GMT ERROR:  duplicate key value violates unique constraint "kpi_assetsnapshot_asset_version_id_key"
> 2018-08-08 14:32:59 GMT DETAIL:  Key (asset_version_id)=(3) already exists.
> 2018-08-08 14:32:59 GMT STATEMENT:  INSERT INTO "kpi_assetsnapshot" ("xml", "source", "details", "owner_id", "asset_id", "_reversion_version_id", "asset_version_id", "date_created", "uid") VALUES ('<?xml version="1.0"?>
> 	<h:html xmlns="http://www.w3.org/2002/xforms" xmlns:ev="http://www.w3.org/2001/xml-events" xmlns:h="http://www.w3.org/1999/xhtml" xmlns:jr="http://openrosa.org/javarosa" xmlns:orx="http://openrosa.org/xforms" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
> 	  <h:head>
> 	    <h:title>tt</h:title>
> 	    <model>
> 	      <instance>
> 	        <data id="snapshot_xml">
> 	          <start/>
> 	          <end/>
> 	          <t1/>
> 	          <meta>
> 	            <instanceID/>
> 	          </meta>
> 	        </data>
> 	      </instance>
> 	      <bind jr:preload="timestamp" jr:preloadParams="start" nodeset="/data/start" type="dateTime"/>
> 	      <bind jr:preload="timestamp" jr:preloadParams="end" nodeset="/data/end" type="dateTime"/>
> 	      <bind nodeset="/data/t1" required="false()" type="string"/>
> 	      <bind calculate="concat(''uuid:'', uuid())" nodeset="/data/meta/instanceID" readonly="true()" type="string"/>
> 	    </model>
> 	  </h:head>
> 	  <h:body>
> 	    <input ref="/data/t1">
> 	      <label>t1</label>
> 	    </input>
> 	  </h:body>
> 	</h:html>
> 	', '{"translated":["label"],"translations":[null],"schema":"1","survey":[{"$autoname":"start","type":"start","name":"start"},{"$autoname":"end","type":"end","name":"end"},{"$autoname":"t1","$kuid":"ha7vm49","required":false,"type":"text","label":["t1"]}],"settings":{"form_title":"tt"}}', '{"status":"success","warnings":[]}', 1, 1, NULL, 3, '2018-08-08T14:32:59.259127+00:00'::timestamptz, 'sLvUQyCRBDoVWtJZBcJubb') RETURNING "kpi_assetsnapshot"."id"
> Aug  8 14:35:01 postgres CRON[210]: (root) CMD (command -v debian-sa1 > /dev/null && debian-sa1 1 1)
>2018-08-08 14:35:45 GMT ERROR:  relation "koboform_surveydraft" does not exist at character 24
>2018-08-08 14:35:45 GMT STATEMENT:  SELECT (1) AS "a" FROM "koboform_surveydraft" LIMIT 1

I've followed all the steps mentioned in the readme file.  I'm using the latest master version of kobo-docker. I analyzed the history of the bug, and I understand that this should be resolved in current versions of kpi and kobocat images. Now when it comes to Enketo, here's what I've verified so far:
- Upon clicking deploy, a request gets sent to 
`https://forms.mydomain.coml/assets/agWfRiwDuuAv8z8hiarWKG/deployment/`
The database contains an entry with such `uid `in the` kpi_asset` table. The `content` column contains what looks like valid json. The response is `{"detail":"No JSON object could be decoded"}`

I've enabled DJANGO_DEBUG for both kpi and kobocat. `docker-compose logs` does not log any errors upon submission. What's more `uwsgi.log` don't contain any errors at all (for both kpi and kobocat). The only trace of the error in in the access log of nginx. 

The one thing which I should probably still do, is sign the certificate because currently it is self-signed. But I feel like this is unrelated, since I can access the rest of application normally.

I'm on Ubuntu 18.04 LTS, Docker 18.03.1-ce, Linux 4.15.0-30-generic.

Please let me know what else could I do to debug this issue, I'll try using a signed certificate tomorrow.

**Comment by kevinholl** [comment]
I have the same errors on my newly built local instance. I am using a signed certificate and the issues persist. Were you able to resolve your issue?

**Comment by Humblehound** [comment]
To some degree, yes. After I've switched to a signed certificate and added entries in docker-compose.yml for extra hosts like described:
```
  extra_hosts:
 - kobocat.domain.com:172.17.3.193
 - koboforms.domain.com:172.17.3.193
```

for all the services, all functionality seems to be working fine. It seems to be important to use the external IP address of the host, not localhost or 0.0.0.0. Let me know if this helps, if not I could share my config with you. 

I'm currently struggling with another issue - I've setup another nginx in front of kobo-docker to setup kobo toolbox on the server with additional services, and while proxy_pass'ing requests seems to be working fine for most stuff, the only functionality which results in the same error is previewing the form inside the editor (it results in the same duplicate key exception). Weirdly deployment and viewing the form is working fine. The verbosity behind these errors makes me want to stay as far away from Django as possible.

**Comment by kevinholl** [comment]
Just closing the loop here, I had to do two things to get my instance working properly:

1.) adding the entire certificate chain to my ssl.crt file in /secrets
2.) adding the extra-hosts in docker-compose.yml. Initially I misconfigured these settings by not properly cross-referencing the containers (e.g. the Enketo container needs hosts for KoboForm and KoboCat, etc.) in the extra-hosts section. Needs to be set very specifically.
